azure_img_utils/cli/image.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+
+"""Azure image utils cli module."""
+
+# Copyright (c) 2022 SUSE LLC
+#
+# This file is part of azure_img_utils. azure_img_utils provides an
+# api and command line utilities for handling images in the Azure Cloud.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+import sys
+
+from azure_img_utils.cli.cli_utils import (
+    add_options,
+    check_required_config_provided,
+    get_config,
+    get_credentials_from_configuration_data,
+    process_shared_options,
+    shared_options,
+    echo_style
+)
+from azure_img_utils.azure_image import AzureImage
+
+
+# -----------------------------------------------------------------------------
+# Image commands function
+@click.group()
+def image():
+    """
+    Commands for image management.
+    """
+
+
+# -----------------------------------------------------------------------------
+# Image blob exists commands function
+@image.command()
+@click.option(
+    '--storage-account',
+    type=click.STRING,
+    required=True,
+    help='Storage account for the blobs.'
+)
+@click.option(
+    '--blob-name',
+    type=click.STRING,
+    required=True,
+    help='Name of the blob to check.'
+)
+@click.option(
+    '--container',
+    type=click.STRING,
+    required=True,
+    help='Container for the blob to check.'
+)
+@add_options(shared_options)
+@click.pass_context
+def blobexists(
+    context,
+    storage_account,
+    blob_name,
+    container,
+    **kwargs
+):
+    """
+    Checks if a image blob exists for the specified container
+    """
+
+    process_shared_options(context.obj, kwargs)
+    config_data = get_config(context.obj)
+    check_required_config_provided(config_data)
+    logger = logging.getLogger('azure_img_utils')
+    logger.setLevel(config_data.log_level)
+    try:
+        credentials = None
+        if (
+            not config_data.sas_token and
+            not config_data.credentials_file
+        ):
+            credentials = get_credentials_from_configuration_data(
```

Comment:
IMHO I think it's better from a user perspective to provide clear instructions how to generate a the proper service principle json than to provide a way for users to copy/paste from that json into another config file. It ends up adding a lot of complexity to code without making the users life any easier.